test/e2e/NonStandardErc20.t.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import {Vm} from "forge-std/Vm.sol";
+
+import {ERC20NoReturn, ERC20ReturningUint} from "../src/NonStandardERC20.sol";
+import {Helper} from "./Helper.sol";
+import {IERC20} from "src/contracts/interfaces/IERC20.sol";
+
+import {GPv2Order, GPv2Signing, SettlementEncoder} from "../libraries/encoders/SettlementEncoder.sol";
+import {Registry, TokenRegistry} from "../libraries/encoders/TokenRegistry.sol";
+
+using SettlementEncoder for SettlementEncoder.State;
+using TokenRegistry for TokenRegistry.State;
+using TokenRegistry for Registry;
+
+contract NonStandardErc20Test is Helper(false) {
+    ERC20NoReturn noReturnToken;
+    ERC20ReturningUint uintReturningToken;
+
+    function setUp() public override {
+        super.setUp();
+
+        noReturnToken = new ERC20NoReturn();
+        uintReturningToken = new ERC20ReturningUint();
```

Comment:
The `ERC20NoReturn()` / `ERC20ReturningUint()` contracts should be brought into this file as they are not used by any other test.